File: stacks_and_queues/backspace_string_compare.js
/**
 * @param {string} s
 * @param {string} t
 * @returns {boolean}
 */
const backspaceCompare = (s, t) => {
    let build = s => {
        let stack = [];
        for (const c of s)
        {
            if (c != "#")
            {
                stack.push(c);
            }
            else if (stack.length)
            {
                stack.pop();
            }
        }

        return stack.join("");
    }

    return build(s) == build(t);
}

File: stacks_and_queues/backspace_string_compare.py
def backspaceCompare(s: str, t: str) -> bool:
    def build(s):
        stack = []
        for c in s:
            if c != "#":
                stack.append(c)
            elif stack:
                stack.pop()
        
        return "".join(stack)
    
    return build(s) == build(t)